sql/catalyst/src/main/scala/org/apache/spark/sql/execution/QueryExecutionException.scala

```diff
@@ -17,5 +17,7 @@
 
 package org.apache.spark.sql.execution
 
+import org.apache.spark.SparkException
+
 class QueryExecutionException(message: String, cause: Throwable = null)
-  extends Exception(message, cause)
+  extends SparkException(message, cause)
```

Comment:
It's good to know that there is a `QueryExecutionException`... We should leave the exception unchanged in PR to avoid unwanted side effects.

Reply:
I moved this to inside `catalyst` so that it would be visible; I don't think this should be problematic as the package is still the same.